tidy3d/components/data/monitor_data.py

```diff
@@ -491,7 +492,7 @@ def mode_area(self) -> FreqModeDataArray:
         num = (intensity * d_area).sum(dim=d_area.dims) ** 2
         den = (intensity**2 * d_area).sum(dim=d_area.dims)
 
-        return FreqModeDataArray(num / den)
+        return FreqModeDataArray(num / den * np.cos(self.monitor.mode_spec.angle_theta))
```

Comment:
Should we check if `isinstance(self.monitor, AbstractModeMonitor)` before accessing `.mode_spec`?

Reply:
Oh I didn't realize this is a property of `ElectromagneticFieldData` and not `ModeSolverData`. I guess in some cases it could be convenient to compute this for a `FieldData` that comes out of the fdtd solver, although it will be kinda up to the user to figure out if it's correct (there can be various instances in which it's not, e.g. if the data does not correspond to single-mode propagation). So in the interest of allowing for that generality, we could leave it there. On the other hand we could move it to `ModeSolverData` to be more strict and make sure that it's always a meaningful quantity? And either way should the polarization fraction computations be treated the same way?